demos/theme/index.js

```diff
@@ -0,0 +1,209 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {HotSwapper, InteractivityProvider, ToolbarProvider} from '../interactivity.js';
+import * as dom from '../dom.js';
+import * as pony from '../ponyfill.js';
+
+const classes = {
+  MDC_LIST_ITEM: 'mdc-list-item',
+  THEME_MENU_ITEM_SELECTED: 'demo-theme-menu__list-item--selected',
+  COMPONENT_SECTION: 'demo-component-section',
+  COMPONENT_SECTION_FLASH: 'demo-component-section--flash',
+  COMPONENT_SECTION_PERMALINK: 'demo-component-section__permalink',
+  COMPONENT_SECTION_HEADING_FOCUS_WITHIN: 'demo-component-section__heading--focus-within',
+};
+
+const attributes = {
+  THEME_NAME: 'data-theme',
+};
+
+const ids = {
+  THEME_STYLESHEET: 'theme-stylesheet',
+  THEME_COLOR_MENU: 'theme-color-menu',
+  THEME_COLOR_MENU_ACTION: 'theme-color-action',
+};
+
+const events = {
+  MDC_SIMPLE_MENU_SELECTED: 'MDCSimpleMenu:selected',
+};
+
+class Permalinker extends InteractivityProvider {
+  /** @param {!Document|!Element} root */
+  static attachTo(root) {
+    const instance = new Permalinker(root);
+    instance.initialize();
+    return instance;
+  }
+
+  /** @private {number} */
+  static get flashDurationMs_() {
+    return 200;
+  }
+
+  /** @override */
+  initialize() {
+    this.root_.addEventListener('focusin', (evt) => {
+      if (evt.target.classList.contains(classes.COMPONENT_SECTION_PERMALINK)) {
+        evt.target.parentElement.classList.add(classes.COMPONENT_SECTION_HEADING_FOCUS_WITHIN);
+      }
+    });
+
+    this.root_.addEventListener('focusout', (evt) => {
+      if (evt.target.classList.contains(classes.COMPONENT_SECTION_PERMALINK)) {
+        evt.target.parentElement.classList.remove(classes.COMPONENT_SECTION_HEADING_FOCUS_WITHIN);
+      }
+    });
+
+    this.window_.addEventListener('hashchange', () => this.flashCurrentSection_());
+    this.flashCurrentSection_();
+  }
+
+  // TODO(acdvorak): Replace page URL hash with current section on scroll
```

Comment:
Reminder that this TODO is here, whether or not you want to deal with it right now; given that we have links in each heading I don't think it's critical.

Reply:
Removed TODO. Thanks!